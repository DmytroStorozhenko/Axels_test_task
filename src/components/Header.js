import React from 'react';
import { Navbar } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';

import Logo from '../assets/img/gunsLogo.png';

import { LoginLink, StyledHeader } from '../styled/componentsStyles/HeaderStyle';

export const Header = () => (
    <>
        <StyledHeader bg="dark" variant="dark">
            <NavLink to="/">
                <Navbar.Brand>
                    <img
                        alt=""
                        src={Logo}
                        width="30"
                        height="30"
                        className="d-inline-block align-top"
                    />{' '}
                    GunsPlace
                </Navbar.Brand>
            </NavLink>
            <LoginLink to={"/login"}>Login</LoginLink>
        </StyledHeader>
    </>
);
