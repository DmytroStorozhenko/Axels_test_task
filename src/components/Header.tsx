import React, { FC } from 'react';
import { Navbar } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';

import { StyledHeader } from '../styled/componentsStyles/HeaderStyle';

import Logo from '../assets/img/gunsLogo.png';

export const Header: FC = () => (
    <>
        <StyledHeader bg="dark" variant="dark">
            <NavLink to="/">
                <Navbar.Brand>
                    <img
                        alt=""
                        src={Logo}
                        width="30"
                        height="30"
                        className="d-inline-block align-top"
                    />{' '}
                    GunsPlace
                </Navbar.Brand>
            </NavLink>
        </StyledHeader>
    </>
);
