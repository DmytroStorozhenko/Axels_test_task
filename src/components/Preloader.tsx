import React, { FC } from 'react';

import { StyledSpinner } from '../styled/componentsStyles/AppStyles';

export const Preloader: FC = () => (
    <>
        <StyledSpinner
            animation="border"
            role="status">
            <span className="sr-only">Загрузка...</span>
        </StyledSpinner>
    </>
);
