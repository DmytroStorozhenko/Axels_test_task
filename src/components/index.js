export { Header } from './Header';
export { ProductItem } from './ProductItem';
export { ProductsList } from './ProductsList';
