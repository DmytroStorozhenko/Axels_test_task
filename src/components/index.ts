export { Header } from './Header';
export { ProductItem } from './ProductItem';
export { ProductsList } from './ProductsList';
export { Preloader } from './Preloader';
export { Rating } from './Raiting';
export { ReviewForm } from './ReviewForm';
