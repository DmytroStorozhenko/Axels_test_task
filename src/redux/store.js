import { createStore, applyMiddleware, combineReducers } from 'redux'
import createSagaMiddleware from 'redux-saga'

import { productsReducer, productsRootSaga } from './productsReducer';

const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers( {
    productsList: productsReducer
} );

export const store = createStore(rootReducer, applyMiddleware( sagaMiddleware ));

sagaMiddleware.run( productsRootSaga )
