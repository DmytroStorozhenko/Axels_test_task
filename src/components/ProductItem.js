import React, { useEffect } from 'react';
import { Col, Container } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { routes } from '../common/routes';
import { addReview, getProduct } from '../redux/productReducer';
import { store } from '../redux/store';

import {
    ArrowBackButton,
    Description,
    Heading,
    ProductInfoContainer,
    ProductItemDescription,
    ProductItemImg,
    ProductRating,
    ProductReviews,
    StyledNavLinkArrow
} from '../styled/componentsStyles/ProductItemStyle';
import { Rating } from './Raiting';
import { ReviewForm } from './ReviewForm';

export const ProductItem = () => {
    const dispatch = useDispatch();
    const { id } = useParams();
    const product = useSelector( () => store.getState().product );
    const { img, name, description, rating, reviews } = product;

    useEffect( () => {
        dispatch( getProduct( id ) )
    }, [dispatch] );

    const submitHandler = (review) => dispatch(addReview([review]))

    return (
        <>
            <Container>
                <StyledNavLinkArrow to={routes.productList}>
                    <ArrowBackButton variant="dark" size="sm">&#60;</ArrowBackButton>
                </StyledNavLinkArrow>
                <ProductInfoContainer lg={11}>
                    <Col sm={10} md={6} lg={6}>
                        <ProductItemImg src={img}/>
                    </Col>
                    <Col sm={10} md={6} lg={6}>
                        <ProductItemDescription>
                            <Heading>{name}</Heading>
                            <Description>{description}</Description>
                        </ProductItemDescription>
                    </Col>
                </ProductInfoContainer>
                <ProductRating lg={10}>
                    <Col>
                        <div>
                            <h4>Оценка: {rating}</h4>
                        </div>
                        <Rating/>
                    </Col>
                </ProductRating>
                <ProductReviews sm={10} lg={12}>
                    <h5>Отзывы:</h5>
                    {!reviews ? '' :
                        reviews.length !== 0 ?
                            reviews.map( (review, index) => (
                                    <p key={index}>{review}</p>
                                )
                            ) :
                            <p>Нет отзывов...</p>
                    }
                    <ReviewForm submitHandler={submitHandler} />
                </ProductReviews>
            </Container>
        </>
    )
};
