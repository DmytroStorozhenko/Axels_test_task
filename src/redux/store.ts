import { createStore, applyMiddleware, combineReducers } from 'redux'
import createSagaMiddleware from 'redux-saga'

import productReducer, { productSaga } from './ducks/product';
import productsListReducer, { productsListSaga } from './ducks/productsList';

const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers( {
    productsList: productsListReducer,
    product: productReducer
} );

export const store = createStore( rootReducer, applyMiddleware( sagaMiddleware ) );

sagaMiddleware.run( productsListSaga )
sagaMiddleware.run( productSaga )

export type AppRootStateType = ReturnType<typeof rootReducer>