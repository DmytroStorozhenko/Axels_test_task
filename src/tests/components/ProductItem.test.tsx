import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { ProductItem } from '../../components';

import { store } from '../../redux/store';

test( 'ProductItem renders correctly', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <BrowserRouter>
                    <ProductItem/>
                </BrowserRouter>
            </Provider>
        )
        .toJSON()
    expect( tree ).toMatchSnapshot()
} );
