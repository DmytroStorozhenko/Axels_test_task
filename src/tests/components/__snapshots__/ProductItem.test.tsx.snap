// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders correctly 1`] = `
<div
  className="container"
>
  <a
    aria-current="page"
    className="sc-gtsrHT sc-pNWdM jqfXrq hIbdhi active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="sc-jrsJWt eFXFIv btn btn-dark btn-sm"
      disabled={false}
      type="button"
    >
      &lt;
    </button>
  </a>
  <div
    className="sc-dlnjwi cbbvwu row row-cols-lg-11"
  >
    <div
      className="col-lg-6 col-md-6 col-sm-10"
    >
      <img
        className="sc-hKFxyN jdAsRn"
        src=""
      />
    </div>
    <div
      className="col-lg-6 col-md-6 col-sm-10"
    >
      <div
        className="sc-eCApnc jSSbuU"
      >
        <h4
          className="sc-jSFjdj"
        >
          
        </h4>
        <p
          className="sc-gKAaRy evMoJY"
        >
          
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-iCoGMd bkQnzH row row-cols-lg-10"
  >
    <div
      className="col"
    >
      <div>
        <h4>
          Оценка: 
        </h4>
      </div>
      <form
        className=""
      >
        <div
          className="align-items-center form-row"
        >
          <div
            className="my-1 col-auto"
          >
            <label
              className="mr-sm-2 form-label sr-only"
              htmlFor="inlineFormCustomSelect"
            >
              Preference
            </label>
            <select
              className="mr-sm-2 custom-select"
              id="inlineFormCustomSelect"
            >
              <option
                value="0"
              >
                ...
              </option>
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
            </select>
          </div>
          <div
            className="my-1 col-auto"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="submit"
            >
              Поставить оценку
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="sc-fujyAs eALNKY row row-cols-lg-12 row-cols-sm-10"
  >
    <h5>
      Отзывы:
    </h5>
    <p>
      Нет отзывов...
    </p>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <textarea
        className="form-control"
        cols={80}
        name="review"
        onBlur={[Function]}
        onChange={[Function]}
        rows={2}
        type="textarea"
        value=""
      />
      <button
        className="sc-bqGGPW dxFzTZ btn btn-primary"
        disabled={false}
        type="submit"
      >
        Оставить отзыв
      </button>
    </form>
  </div>
</div>
`;

exports[`Snapshot for ProductItem component 1`] = `
<div
  className="container"
>
  <a
    aria-current="page"
    className="sc-gtsrHT sc-pNWdM jqfXrq hIbdhi active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="sc-jrsJWt eFXFIv btn btn-dark btn-sm"
      disabled={false}
      type="button"
    >
      &lt;
    </button>
  </a>
  <div
    className="sc-dlnjwi cbbvwu row row-cols-lg-11"
  >
    <div
      className="col-lg-6 col-md-6 col-sm-10"
    >
      <img
        className="sc-hKFxyN jdAsRn"
        src=""
      />
    </div>
    <div
      className="col-lg-6 col-md-6 col-sm-10"
    >
      <div
        className="sc-eCApnc jSSbuU"
      >
        <h4
          className="sc-jSFjdj"
        >
          
        </h4>
        <p
          className="sc-gKAaRy evMoJY"
        >
          
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-iCoGMd bkQnzH row row-cols-lg-10"
  >
    <div
      className="col"
    >
      <div>
        <h4>
          Оценка: 
        </h4>
      </div>
      <form
        className=""
      >
        <div
          className="align-items-center form-row"
        >
          <div
            className="my-1 col-auto"
          >
            <label
              className="mr-sm-2 form-label sr-only"
              htmlFor="inlineFormCustomSelect"
            >
              Preference
            </label>
            <select
              className="mr-sm-2 custom-select"
              id="inlineFormCustomSelect"
            >
              <option
                value="0"
              >
                ...
              </option>
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
            </select>
          </div>
          <div
            className="my-1 col-auto"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="submit"
            >
              Поставить оценку
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="sc-fujyAs eALNKY row row-cols-lg-12 row-cols-sm-10"
  >
    <h5>
      Отзывы:
    </h5>
    <p>
      Нет отзывов...
    </p>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <textarea
        className="form-control"
        cols={80}
        name="review"
        onBlur={[Function]}
        onChange={[Function]}
        rows={2}
        type="textarea"
        value=""
      />
      <button
        className="sc-bqGGPW dxFzTZ btn btn-primary"
        disabled={false}
        type="submit"
      >
        Оставить отзыв
      </button>
    </form>
  </div>
</div>
`;
